client/kie-client/src/main/java/org/apache/servicecomb/kie/client/KieRawClient.java

```diff
@@ -0,0 +1,188 @@
+
+  private static final String DOMAIN_NAME = "default";
+
+  private static final String V4_PREFIX = "v1";
+
+  private String basePath;
+
+  private String host;
+
+  private int port;
+
+  private String domainName;
+
+  private HttpTransport httpTransport;
+
+  public KieRawClient() {
+    this(DEFAULT_HOST, DEFAULT_PORT, DOMAIN_NAME, HttpTransportFactory.getDefaultHttpTransport());
+  }
+
+  private KieRawClient(String host, int port, String domainName, HttpTransport httpTransport) {
+    this.host = host;
+    this.port = port;
+    this.domainName = domainName;
+    this.httpTransport = httpTransport;
+
+    // check that host has scheme or not
+    String hostLowercase = host.toLowerCase();
+    if (!hostLowercase.startsWith("https://") && !hostLowercase.startsWith("http://")) {
+      // no protocol in host, use default 'http'
+      host = "http://" + host;
+    }
+
+    this.basePath = host + ":" + port + "/" + V4_PREFIX + "/" + domainName;
+  }
+
+  public HttpResponse getHttpRequest(String url, Map<String, String> headers, String content) throws IOException {
+
+    if (headers == null) {
+      headers = new HashMap<String, String>();
+    }
+
+    HttpRequest httpRequest = new HttpRequest(basePath + url, headers, content);
+
+    return httpTransport.get(httpRequest);
+  }
+
+  public HttpResponse postHttpRequest(String url, Map<String, String> headers, String content) throws IOException {
+
+    if (headers == null) {
+      headers = new HashMap<String, String>();
+    }
+
+    HttpRequest httpRequest = new HttpRequest(basePath + url, headers, content);
+
+    return httpTransport.post(httpRequest);
+  }
+
+  public HttpResponse putHttpRequest(String url, Map<String, String> headers, String content) throws IOException {
+
+    if (headers == null) {
+      headers = new HashMap<String, String>();
+    }
+
+    HttpRequest httpRequest = new HttpRequest(basePath + url, headers, content);
+
+    return httpTransport.put(httpRequest);
+  }
+
+  public HttpResponse deleteHttpRequest(String url, Map<String, String> headers, String content) throws IOException {
+
+    if (headers == null) {
+      headers = new HashMap<String, String>();
+    }
+
+    HttpRequest httpRequest = new HttpRequest(basePath + url, headers, content);
+
+    return httpTransport.delete(httpRequest);
+  }
+
+  public HttpTransport getHttpTransport() {
+    return httpTransport;
+  }
+
+  public void setHttpTransport(HttpTransport httpTransport) {
+    this.httpTransport = httpTransport;
+  }
+
+  public static class Builder {
+    private String host;
+
+    private int port;
+
+    private String domainName;
+
+    private HttpTransport httpTransport;
+
+    public Builder() {
+      this.host = DEFAULT_HOST;
+      this.port = DEFAULT_PORT;
+      this.domainName = DOMAIN_NAME;
+    }
+
+    public String getDomainName() {
+      return domainName;
+    }
+
+    public Builder setDomainName(String domainName) {
+      if (domainName == null) {
```

Comment:
I think delete this code is better. Port, Host is the same. [minor]

Reply:
The domainName means projectName, mayebe this naming is easy to misunderstand. I can change the name to projectName.
The base api path is "/v1/{project}/kie/kv/" in swagger. The default value of {project} is default, which can also be customized by the user.